src/useCameraShake.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react'
+import * as THREE from 'three'
+import { useFrame, Camera } from 'react-three-fiber'
+import { makeNoise2D } from 'open-simplex-noise'
+
+type ShakeConfig = typeof defaultConfig
+export type ShakeConfigPartial = Partial<ShakeConfig>
+
+const defaultConfig = {
+  decay: false,
+  decayRate: 0.65,
+  maxYaw: 0.1,
+  maxPitch: 0.1,
+  maxRoll: 0.1,
+  yawFrequency: 1,
+  pitchFrequency: 1,
+  rollFrequency: 1,
+  yawNoiseSeed: 10,
+  pitchNoiseSeed: 20,
+  rollNoiseSeed: 30,
+}
+
+export interface ShakeController {
+  setTrauma: (val: number) => void
+  getTrauma: () => number
+  addTrauma: (val: number) => void
+  clearTrauma: () => void
+}
+
+type ShakeCamera = Camera & ShakeController
+
+export function useCameraShake(
+  baseCamera: Camera | undefined,
+  shakeConfig?: ShakeConfigPartial,
+  initialTrauma: number = 1
+) {
+  const cameraRef = React.useRef<Camera>(new THREE.PerspectiveCamera())
```

Comment:
Because of what we're doing, this should be `THREE.Camera` not the r3f Camera type.